### PR TITLE
Introducing management services in Fundamentals

### PR DESCRIPTION
**Pull request by einari**
This pull request has a couple of things: 

* Moving management service discovery and definition into Fundamentals to be able to provide management services for fundamental building blocks
* Introducing a first management service for dependency inversion - getting bindings - proves the system and provides valuable insight
* Additions to the configuration system for being able to provide default instances of configuration objects. This allows us to be backwards compatible and not having to bump major versions right now. Plus it is valuable to be able to provide default configuration objects if no provider can provide a configuration object.

**Notes:**
The configuration system carries with it a technical debt of not having specs. This is defined as an issue in #251

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1137979265727616)

### CODE REVIEW

File: Protobuf/system/object.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+syntax = "proto3";
+
+package dolittle.events.relativity;
+
+
+option csharp_namespace = "System.Protobuf";
+
+// Represents the artifact
```
Comment (woksin) [inline Protobuf/system/object.proto]: Both object.proto and value.proto "Represents the artifact"

Comment (woksin) [reply Protobuf/system/object.proto]: I can't seem to follow

Comment (einari) [reply Protobuf/system/object.proto]: Good catch - both files will be gone in a future commit in a completely different pull request on a different branch :)

**Comment by einari** [comment]
The proto files we have now are .NET Fundamentals specific, for now. That might change in the future - and then we can promote them I guess. This is the first stab at all this, aiming for future clarity :). My thinking has been for versioning sake that the thing exposing something maintains the proto files as well. Adhering to the high cohesion principle. But where it gets a bit weird is who is responsible for creating the language specific code-generated proxies. I’ve put this responsibility for now onto the owner of the endpoint and proto files, which in one way feels a bit weird - but the nice benefit of it is that the language specific packages coming out is semantically versioned according to the source it represents. Which makes it a lot simpler for us right now at least.

What I was missing was a way to agnostically package proto files, version these packages and publish them to a “proto package library” and others could then have dependencies to this in a versioned manner.

**Comment by jakhog** [comment]
Hmm, yeah that's a good point. Things like IoC bindings doesn't necessarily make any sense in other languages. And yes, it feels weird - but I agree that this responsibility should be on the owner of the endpoint.

**Pull request merged**